src/Microsoft.Diagnostics.Monitoring.Egress/FileSystem/FileSystemEgressEndpoint.cs

```diff
@@ -0,0 +1,65 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.Diagnostics.Monitoring.Egress.FileSystem
+{
+    internal class FileSystemEgressEndpoint : IEgressEndpoint<FileSystemEgressStreamOptions>
+    {
+        private readonly FileSystemEgressEndpointOptions _endpointOptions;
+
+        public FileSystemEgressEndpoint(FileSystemEgressEndpointOptions endpointOptions)
+        {
+            _endpointOptions = endpointOptions;
+        }
+
+        public async Task<EgressResult> EgressAsync(
+            Func<CancellationToken, Task<Stream>> action,
+            string name,
+            FileSystemEgressStreamOptions streamOptions,
+            CancellationToken token)
+        {
+            using var fileStream = CreateFileStream(name, out string filePath);
+
+            using Stream stream = await action(token);
```

Comment:
I'm wondering if there's some ways to unify these two implementations:
- Use a base class instead of interface for egress
- Declare an abstract api that has the Func<Stream, CancellationToken, Task> action parameter
- Declare a virtual function that takes the Func<CancellationToken, Task<Stream>> action,
- This implementation could just call the abstract method with the CopyToAsync as the method body.

Reply:
Done